### ISSUE TITLE
cannot reference slider in onSliderLoad event

### ISSUE THREAD

**Issue by jmp909** [stevenwanderski/bxslider-4#826]
the code below doesn't work. `_slider` is not recongnised as a bxslider object

it seems like we have to call a timeout delay after onSliderLoad before `_slider`  exists. i'd be concerned that on slower machines this value would need to be higher

thanks
j

http://jsbin.com/fizoxazizo/

```
$(window).load(function(){
  var _slider; 
  _slider = $('.bxslider').bxSlider({
    mode: 'horizontal',
    preload: 'visible',
    onSliderLoad: function(currentIndex) {
        _slider.gotoSlide(2);                
    }
  });
});
```

**Comment by jmp909** [comment]
this version does work.. with the timeout as 0. but you can't call the function immediately eg doFinish(), it has to be on the timeout

```
var _slider; 

$(window).load(function(){
  
  
  _slider = $('.bxslider').bxSlider({

    mode: 'horizontal',
    preload: 'visible',

    onSliderLoad: function(currentIndex) {

      window.setTimeout(doFinish,0);
        
    }
  });
});

function doFinish() {
  _slider.goToSlide(2);
}
```

**Comment by Tidal-Wave** [comment]
please note that you are linking to a very old version of bxSlider from the site. I currently don't control the site and can't change the version found there. Please first try using the most recent version found here. If you still have an issue we can work through it. Note that in current versions the callbacks have changed slightly to include references to the object, which may help. I also recently implemented stored references using `.data()`

**Comment by jmp909** [comment]
thanks. i missed this a couple of days ago. I've got the new version working now. I had to modify my code slightly to use a _settings variable so i could do a reloadSlider(_settings) on reisize otherwise my functionality was being lost because it wasn't keeping the onSliderLoad and onSlideBefore custom functions when i call reloadSlider
all good now though.
regardsj

Date: Tue, 2 Jun 2015 06:49:47 -0700
